rest-backend/src/test/java/com/sap/psr/vulas/backend/rest/BugControllerTest.java

```diff
@@ -362,6 +386,92 @@ public void testPutNotFound() throws Exception {
+    	mockMvc.perform(post_builder)	
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$.wellknownDigest", is(true)))
+                .andExpect(jsonPath("$.digest", is("1E48256A2341047E7D729217ADEEC8217F6E3A1A")));
+    	
+
+    	// Repo must contain 1
+    	assertEquals(1, this.libRepository.count());
+    	
+    	LibraryId lid = new LibraryId("com.foo", "bar", "0.0.1");
+    	
+    	libIdRepository.save(lid);
+    	
+    	ConstructId cid = cidRepository.FILTER.findOne(cidRepository.findConstructId(ProgrammingLanguage.JAVA, ConstructType.CLAS, "com.acme.Foo"));
+    	ConstructChange cc = ccRepository.FILTER.findOne(ccRepository.findByRepoPathCommitCidBug("svn.apache.org", "/trunk/src/main/java/com/acme/Foo.java", "123456", cid, bug));
+    	
+    	
+    	AffectedLibrary[] afl = (AffectedLibrary[])JacksonUtil.asObject(FileUtil.readFile(Paths.get("./src/test/resources/real_examples/affectedLib-propagate.json")), AffectedLibrary[].class);
+    	
+    	AffectedConstructChange acc = new AffectedConstructChange(cc, afl[0], true, true, true, null);
+    	Collection<AffectedConstructChange> accList = new ArrayList<AffectedConstructChange>();
+    	accList.add(acc);
+    	afl[0].setAffectedcc(accList);
+    	
+    	post_builder = post("/bugs/CVE-2014-0050/affectedLibIds?source=PROPAGATE_MANUAL")
+    			.content(JacksonUtil.asJsonString(afl).getBytes())
+				.contentType(MediaType.APPLICATION_JSON)
+				.accept(MediaType.APPLICATION_JSON);
+    	mockMvc.perform(post_builder)	
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$.length()", is(2)));
+    	
+    	
+    	MockHttpServletRequestBuilder put_builder = put("/bugs/CVE-2014-0050/affectedLibIds?source=PROPAGATE_MANUAL")
+    			.content(JacksonUtil.asJsonString(afl).getBytes())
+				.contentType(MediaType.APPLICATION_JSON)
+				.accept(MediaType.APPLICATION_JSON);
+    	mockMvc.perform(put_builder)	
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(contentType));
+    	
+
+    	MockHttpServletRequestBuilder get_builder = get("/bugs/CVE-2014-0050/affectedLibIds/bar/bar/0.0.1?source=PROPAGATE_MANUAL");
+    	mockMvc.perform(get_builder)	
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(contentType))
+        .andExpect(jsonPath("$.[0].affectedcc.length()", is(1)));
+    	
+    	afl[0].setAffected(null);
+    	
+    	put_builder = put("/bugs/CVE-2014-0050/affectedLibIds?source=PROPAGATE_MANUAL")
+    			.content(JacksonUtil.asJsonString(afl).getBytes())
+				.contentType(MediaType.APPLICATION_JSON)
+				.accept(MediaType.APPLICATION_JSON);
+    	mockMvc.perform(put_builder)	
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(contentType))
+        .andExpect(jsonPath("$.[0].affected").value(IsNull.nullValue()));
```

Comment:
Would it be possible to check whether `modified_at` was properly updated?

Reply:
Done in https://github.com/SAP/vulnerability-assessment-tool/pull/68/commits/ede57c9bdaecdfd826485b4dcc83f6614d6d7250